### PR TITLE
Added base64 to UIImageExtensions

### LINKED ISSUE #355: UIImage.base64
Nowadays we have an extension that initializes an ```UIImage``` from a base64 String, but we don't have a way to easily convert an UIImage to base64

I'm currently in a project so I don't have time to open the PR. Will do it tomorrow
``` swift
extension UIImage {
    
    var base64: String {
        
        let data = UIImageJPEGRepresentation(self, 1.0)
        return data!.base64EncodedString()
    }
}
```

### PR DESCRIPTION
**Pull request by lfarah**
fixes #355 
## Checklist

- [x] New Extension
- [ ] New Test
- [ ] Changed more than one extension, but all changes are related
- [ ] Trivial change (doesn't require changelog)

### CODE REVIEW

File: Sources/UIImageExtensions.swift [hunk 1]
```diff
@@ -9,6 +9,12 @@
 import UIKit
 
 extension UIImage {
+    
+    /// EZSE: Returns base64 string
+    var base64: String {
+        return UIImageJPEGRepresentation(self, 1.0)!.base64EncodedString()
```
Comment (Khalian) [inline Sources/UIImageExtensions.swift]: Is it possible to test this ?

Comment (Khalian) [inline Sources/UIImageExtensions.swift]: I would avoid network IO if I could. Lets keep a downloaded image in some testdata folder or something of that order. Or just on charizard.png.

Comment (lfarah) [inline Sources/UIImageExtensions.swift]: @Khalian couldn't get the image to work. Keep getting nil

Comment (Khalian) [inline Sources/UIImageExtensions.swift]: @lfarah Which image did you test it on ?

Comment (lfarah) [inline Sources/UIImageExtensions.swift]: The charizard one. Can you try to do it?

Comment (Khalian) [inline Sources/UIImageExtensions.swift]: Yes I was able to get it to work. My prototype code was : 

    public func testImageBase64() {
        let image = UIImage(contentsOfFile: "/workplace/arunavs/third_party/EZSwiftExtensions/charizard.png")
        print(image?.base64)
    }

It gave a non nil base 64 encoded string. 

Can you write a test(with the path formatted as a relative path) to assert that the output is not nil and also its of base 64 format (do we have an extension/foundation API for that?). I will approve this after that.

**Comment by lfarah** [comment]
@Khalian I have no idea why the test is not working. Converted the .png to .jpg and the base64 string looks visually the same as the test str I got [from this website](http://base64image.org/)

**Comment by Khalian** [comment]
You can avoid the merge commits by rebasing the branch locally and force pushing onto the remote branch (never force push on mainline).

**Comment by lfarah** [comment]
I can clean it later... Just wanna get this test working

**Comment by lfarah** [comment]
@vsouza can you help?

**Comment by Khalian** [comment]
As long as you get some base64 string, IMO its good enough a test.

**Comment by lfarah** [comment]
@Khalian what do you think?

File: EZSwiftExtensionsTests/UIImageTests.swift [hunk 2]
```diff
@@ -7,10 +7,16 @@
...
 
-    
-
+    func testBase64() {
+        let testBundle = Bundle(for: type(of: self))
+        let imageURL = testBundle.url(forResource: "charizard", withExtension: "jpg")
+        let dataImage = try? Data(contentsOf: imageURL!)
+        let image = UIImage(data: dataImage!)
+        
+       XCTAssertNotNil(image!.base64)
```
Comment (Khalian) [inline EZSwiftExtensionsTests/UIImageTests.swift]: Lets add a stronger check here:-

if let decodedData = NSData(base64EncodedString: image!.base64, options: nil) {
    // Pass test case
} else {
   // Fail test case.     
}

**Comment by Khalian** [comment]
Dont forget to squash the commits into one.

**Comment by Khalian** [comment]
@piv199 @goktugyil What do you guys think ?

**Comment by Khalian** [comment]
@lfarah Can you replay your changes on existing master ?

**Comment by Khalian** [comment]
@goktugyil Its a base64 conversion of an image. Traditionally base64 conversions for different inputs (including binary encodings) are used to transmit information over the HTTP protocol.

**Comment by goktugyil** [comment]
I understand that part :D

**Comment by vsouza** [comment]
@goktugyil Basically base64 is the most used way to do image upload. I use it all the time on my freelance projects. There are a lot of changes because testing it was really shitty

**Comment by lfarah** [comment]
@goktugyil Basically base64 is the most used way to do image upload. I use it all the time on my freelance projects. There are a lot of changes because testing it was really shitty

**Pull request merged**